vertx-web/src/main/java/io/vertx/ext/web/impl/RouteImpl.java

```diff
@@ -232,6 +233,18 @@ synchronized boolean matches(RoutingContext context, String mountPoint, boolean
     }
     HttpServerRequest request = context.request();
     if (!methods.isEmpty() && !methods.contains(request.method())) {
+      context.response().putHeader(HttpHeaders.ALLOW, String.join(", ", () -> new Iterator<CharSequence>() {
```

Comment:
Loss of performance. If some fail to match but a specific one ends up being matched, all this work to build the `Allow` header becomes dead work. Same happens if many fail to match